### PR TITLE
feat: add GitHub workflows

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Add GitHub workflows to test PR changes and build/push Docker images once the PR is merged to main.

Also includes linting and formatting fixes for the projects.

# Related
- cds-snc/platform-core-services#175

### CODE REVIEW

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+jobs:
+  docker-build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        lambda:
+          - blazer
+          - google-cidr
```
Comment (jimleroyer) [inline .github/workflows/docker-build.yml:17]: So one docker to rule them all? (i.e. on docker image for all lambdas)

File: .github/workflows/docker-build.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          filters: |
+            lambda:
+              - '${{ matrix.lambda }}/**'
+              - '.github/workflows/docker-build.yml'
+
+      - name: Build Docker image
+        if: steps.changes.outputs.lambda == 'true'
+        working-directory: ${{ matrix.lambda }}
+        run: make docker
```
Comment (jimleroyer) [inline .github/workflows/docker-build.yml:34]: To which Makefile does this refer to?

File: .github/workflows/test-format-lint.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        if: steps.changes.outputs.lambda == 'true' && matrix.ruby
+        uses: ruby/setup-ruby@c4fe7bd15ddbfcd5e07e47bf2f2cae90581d6091 # tag=v1.123.0
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+
+      - name: Install deps
+        if: steps.changes.outputs.lambda == 'true'
+        working-directory: ${{ matrix.lambda }}
+        run: make install
```
Comment (jimleroyer) [inline .github/workflows/test-format-lint.yml:50]: To which Makefile does this refer to?

File: bin/scaffold.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+install:
+	echo "::warning file=blazer/Makefile,line=14,col=1::install is not setup..."
+
+lint:
+	echo "::warning file=blazer/Makefile,line=14,col=1::lint is not setup..."
+
+test:
+	echo "::warning file=blazer/Makefile,line=14,col=1::test is not setup..."
+
+.PHONY: default	docker fmt install lint	test
```
Comment (jimleroyer) [inline bin/scaffold.sh:31]: Ah there is one Makefile for each lambda, and that's the scaffold. 👍

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (patheard) [reply .github/workflows/docker-build.yml:17]: Nope, each Lambda will have its own Dockerfile in its folder.  That'll give us the flexibility to tailor them to the specific Lambda's needs and language.

File: blazer/Makefile [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+test: # todo: write controller and user model tests
+	echo "::warning file=blazer/Makefile,line=17,col=1::test is not setup..."
+
+.PHONY: \
+	default \
+	docker \
+	fmt \
+	install	\
+	lint \
+	test
```
Comment (jimleroyer) [inline blazer/Makefile:26]: Ok I don't know why but I didn't see that Makefile when I chrome search previously. 🙈

**Review by jimleroyer — APPROVED** [review]
LGTM

File: .github/workflows/docker-build.yml (hunk 2 above)
Comment (patheard) [reply .github/workflows/docker-build.yml:34]: The `matrix` job setup means that at this point, the `working-directory: ${{ matrix.lambda }}` line will be executing the `Makefile` within the lambda's folder.

When a new lambda gets added, it'll just be a matter of adding its folder to the matrix like so:

```yaml
matrix:
    lambda:
        - blazer
        - google-cidr
        - a-new-and-highly-fancy-lambda
```

**Review by jzbahrai — APPROVED** [review]

**Pull request merged**